cli/src/semgrep/commands/scan.py

```diff
@@ -312,6 +312,12 @@ def convert(
     ),
     optgroup.option("--dump-command-for-core", "-d", is_flag=True, hidden=True),
     optgroup.option("--allow-untrusted-postprocessors", is_flag=True, hidden=True),
+    optgroup.option(
+        "--no-secrets-validation",
+        "disable_secrets_validation_flag",
```

Comment:
Why do we have this as a parameter decl?

Reply:
I'm not sure what a parameter decl is, I added this based on the PR you sent me in my attempt to move this from `ci` to `scan`

Reply:
I'm just curious about the declaration of the `disable_secrets_validation_flag` parameter. I don't think there's anything like that in the PR I sent you, and if I understand the click docs correctly, this is going to declare an additional parameter/command line flag with that name (which I don't think you want).

Reply:
Is it additional or just choosing the name of the parameter in the python function? My read of the documentation is latter? In which case I think this is fine. It might not be necisarry. I think he is copying my work on the secrets flag which I added the "_flag" bit to the name, because I thought it was confusing that you might not specify the flag but still get secrets enabled because of scan_handler has it enabled.